lib/tty/command/process_runner.rb

```diff
@@ -169,6 +166,15 @@ def read_stream(stream, buffer)
         end
       end
 
+      def next_written_line(stream)
+        return stream.gets unless @timeout
+
+        limit = (Thread.current[:cmd_start] + @timeout) - Time.now
+        Timeout.timeout(limit) { return stream.gets }
```

Comment:
Generally speaking I stay away from using `Timeout` library as it is known not be thread safe and cause buggy code [read more on Mike Perham's blog](http://www.mikeperham.com/2015/05/08/timeout-rubys-most-dangerous-api/). I would like to see if there is a better way to track time and timeout. For example, `select` accepts as an argument timeout which I'm using when actually hooking up all the pipes.